### PR TITLE
Reenable arm helix queues

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

**Review by dougbu — APPROVED** [review]

**Review by Eilon — APPROVED** [review]

**Review by Eilon — APPROVED** [review]

**Review by Eilon — APPROVED** [review]

**Comment by HaoK** [comment]
Looks like the scripts we use to bootstrap dotnet to run our tests don't work in the docker helix environment (no curl is the first issue), this is going to require more work to enable

**Comment by HaoK** [comment]
Also for windows arm queue the scripts are still trying to install/run the x64 dotnet

File: .azure/pipelines/helix-test.yml
Comment (dougbu) [inline .azure/pipelines/helix-test.yml]: This needs ` -arch x64` doesn't it?

**Review by dougbu — APPROVED** [review]
We don't have Linux on ARM agents and pools yet? Or, we don't really need to test there as well?

**Comment by HaoK** [comment]
https://github.com/dotnet/corefx/blob/10d7d6f812295a572eb148aedb19efd2f96fbe17/eng/pipelines/windows.yml is how they split things up

```
            x64_Release:
              _BuildConfig: Release
              _architecture: x64
              _framework: netcoreapp
              _helixQueues: $(netcoreappWindowsQueues)+$(nanoQueues)
              _publishTests: true

            x86_Release:
              _BuildConfig: Release
              _architecture: x86
              _framework: netcoreapp
              _helixQueues: $(netcoreappWindowsQueues)
              _publishTests: true

            arm64_Release:
              _BuildConfig: Release
              _architecture: arm64
              _framework: netcoreapp
              _helixQueues: $(windowsArmQueue)
              _publishTests: true
```

File: eng/targets/Helix.Common.props
Comment (natemcmaster) [inline eng/targets/Helix.Common.props]:
```suggestion
        <HelixAvailibleTargetQueue Include="Windows.10.Arm64.Open" Platform="Windows"  Arch="arm64" />
```

File: eng/targets/Helix.Common.props
Comment (JunTaoLuo) [inline eng/targets/Helix.Common.props]: Can we also add a Alpine ARM64 Queue? https://github.com/dotnet/corefx/blob/56115730ea260dd27cf74818dff858cb8424e61a/eng/pipelines/linux.yml#L122

I just added a linux-musl-arm64 configuration for aspnetcore.

**Comment by HaoK** [comment]
Remaining issues: need to pick a different destination to install cli, docker can't write to correlation directory. Also helix.proj fails with an error if no work items are sent, so need to filter differently

**Comment by natemcmaster** [comment]
> need to pick a different destination to install cli, docker can't write to correlation directory

FYI - I'm addressing this in https://github.com/aspnet/AspNetCore/pull/10409

File: eng/targets/Helix.Common.props
Comment (natemcmaster) [inline eng/targets/Helix.Common.props]:
```suggestion
    <HelixAvailableTargetQueue Include="(Debian.9.Arm64.Open)Ubuntu.1604.Arm64.Docker.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:debian-9-helix-arm64v8-0a0ebdd-20190312215438" Platform="Linux" Arch="arm64" />
```

This doesn't seem to match. The queue name says Arm64, but you're running "arm" builds on it (arm32). Should this be corrected?

File: .azure/pipelines/helix-test.yml
Comment (natemcmaster) [inline .azure/pipelines/helix-test.yml]:
```suggestion
    - script: .\build.cmd -arch arm64 -all -ci /t:Helix /p:IsHelixJob=true /bl:$env:BUILD_SOURCESDIRECTORY\artifacts\logs\SendToHelixArm.binlog
```

**Review by natemcmaster — COMMENTED** [review]
Other than mismatched arm architectures, changes look fine.

File: .azure/pipelines/helix-test.yml
Comment (HaoK) [reply .azure/pipelines/helix-test.yml]: This is the error from trying arm64: Cannot validate argument on parameter 'Architecture'. The argument "arm64" does not belong to the set "x64,x86,arm"

Comment (natemcmaster) [reply .azure/pipelines/helix-test.yml]: Cc @JunTaoLuo

I don't think we actually support building for Windows Arm64 yet, only Linux Arm64. That's why you're getting this error. Maybe this should build on Linux instead?

Comment (JunTaoLuo) [reply .azure/pipelines/helix-test.yml]: Correct we don't have win-arm64 yet.

**Comment by HaoK** [comment]
Finally progress, all tests except 4 that are doing custom copying are passing on Debian.9.Arm64.  The Ubuntu arm queue is failing because curl isn't found :/

**Comment by HaoK** [comment]
@MattGal is there any way we can get curl installed on the Arm64 docker queue?

`"(Ubuntu-18.04.Arm64.Open)Ubuntu.1604.Arm64.Docker.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-0a0ebdd-20190312220351`

**Comment by MattGal** [comment]
@HaoK actually yeah that's super easy, since mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-0a0ebdd-20190312220351 is a docker image.  

Just make a PR here:  https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/master/src/ubuntu/18.04/helix/arm64v8/Dockerfile .  Once it merges and a subsequent build runs, an updated tag like what you're using will appear on the list of tags served up by the repo, and you can use that immediately in your work items.

I am not a linux expert, but I think it may literally be as easy as here:
https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/148
(and the CI build will confirm if this works, as I don't actually have such a machine to **docker build** on)

@MichaelSimons is the SME here for repo-specific questions as well.

**Comment by HaoK** [comment]
Awesome thanks @MattGal yeah sounds like curl is part of the default ubuntu repositories, and just needs the equivalent of `sudo apt install curl` which looks like what your PR adds

**Comment by MattGal** [comment]
https://dev.azure.com/dnceng/public/_build/results?buildId=213875&view=logs is green for my change, so we just need to get a signoff from the right folks and should be usable within 30 minutes of merge.

**Comment by MattGal** [comment]
@HaoK following up from yesterday, you should now be able to use queue `"(Ubuntu-1804.Arm64.Open)Ubuntu.1604.Arm64.Docker.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-6f28fa9-20190606004102`  (note my deletion of a period in the friendly-reporting name just to match how other Helix queues are named)

**Comment by HaoK** [comment]
Thanks @MattGal I didn't realize until now that I have to update the queue name to see your changes, duh, trying it now

**Comment by HaoK** [comment]
@MattGal seeing some new chmod errors I haven't seen before on the new ubuntu queue:

https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190607.16/workItem/Microsoft.AspNetCore.NodeServices.Tests~2Fnetcoreapp3.0/wilogs

```
[BEGIN EXECUTION]
chmod: changing permissions of '/root/helix/work/workitem/installnode.sh': Operation not permitted
chmod: changing permissions of '/root/helix/work/workitem/runtests.sh': Operation not permitted
+ ./installnode.sh 10.15.3 arm64
/root/helix/work/correlation/scripts/4bfc2b9d80b5462fbe3d2a887ddfd6e6/execute.sh: 4: /root/helix/work/correlation/scripts/4bfc2b9d80b5462fbe3d2a887ddfd6e6/execute.sh: ./installnode.sh: Permission denied
```

Any ideas what might be causing this?  The Debian docker queue seems fine running the same scripts

**Comment by MattGal** [comment]
@HaoK likely we need to tweak stuff about the users in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/master/src/ubuntu/18.04/arm64v8/Dockerfile ... @ulisesh  may already know what's needed here, I mostly would need an ARM64 machine to investigate it myself.

**Pull request closed without merging**
